src/Reference.cc

```diff
@@ -43,4 +43,8 @@ std::vector<std::string> Local<Object>::getKeyNames() const {
   return ret;
 }
 
+bool Local<Number>::isInteger() const {
+  return num.asNumber().toDouble() - num.asNumber().toInt64() < 1e-15;
```

Comment:
This is some magic code, Lua(also JS, etc) doesn't really provide Integer type.
If you really need to check this, do it in your on project code.